runner/src/index.ts

```diff
@@ -55,3 +57,61 @@ export function test() {
 
   return 0;
 }
+
+const matchIdenticalTopLevel = (output: any) => {
```

Comment:
what if instead of doing all this we just have a generic deepequal. we parse the result and then deepequal it to the kitchen sink.

Something like this:

```javascript

function deepEqual(obj1, obj2) {
    if (obj1 === obj2) {
        return true;
    }

    if (obj1 == null || obj2 == null || typeof obj1 !== 'object' || typeof obj2 !== 'object') {
        return false;
    }

    let keys1 = Object.keys(obj1);
    let keys2 = Object.keys(obj2);

    if (keys1.length !== keys2.length) {
        return false;
    }

    for (let key of keys1) {
        if (!keys2.includes(key) || !deepEqual(obj1[key], obj2[key])) {
            return false;
        }
    }

    return true;
}

Test.assertTrue(
    "reflectJsonObject preserved the KitchenSink JSON object",
    deepEqual(output, KitchenSink))
```

Reply:
trying to modify it a bit to also support the less strict but still _accurate enough_ Date testing... but a negative consequence of this is that its impossible to tell which aspect of the test/object _doesn't_ pass... unless we add the assertions inside the `deepEqual` function instead.

Reply:
@bhelx - I couldn't get this to work. If you feel like its a more comprehensive approach, would you want to hold off on this PR so you can explore it further? 

I have at least enough confidence now in the Go bindgen implementation to move forward with that PR.

Reply:
yeah i can come back and revisit. feel free to just focus on getting it passing the tests for you! it's not a blocker or anything.

Reply:
ok, I've manually tested both Go (with `date-time` support) and Typescript (on its `main`) branch agains these new tests and all things pass.